src/c_api/c_api_function.cc

```diff
@@ -107,6 +107,24 @@ void Backward(const OpStatePtr& state,
   Imperative::Get()->set_is_recording(prev_recording);
 }
 
+// infer storage function for custom op, which assigns kDefaultStorage for
+// all undefined stypes, and dispatch on kDispatchFComputeEx.
+inline bool CustomFunctionStorageType(const nnvm::NodeAttrs& attrs,
+                                      const int dev_mask,
+                                      int* dispatch_mode,
+                                      std::vector<int> *iattr,
+                                      std::vector<int> *oattr) {
+  for (int& v : *oattr) {
```

Comment:
Why int by reference here?  This means indirect addressing when accessed which is slower than just holding the scalar value.
ie:
for example, indirect addressing:
  lea esi, [oattr + 4]  ; for example, second item
  mov eax, [esi]  ; eax now holds 

as opposed to scalar immediate:
  mov eax, [oattr + 4]

Reply:
Well I still need to write to that address in the for loop. Does it really harm?